llvm/lib/Target/PowerPC/PPCISelDAGToDAG.cpp

```diff
@@ -7780,6 +7775,10 @@ void PPCDAGToDAGISel::PeepholePPC64() {
       Flags = PPCII::MO_TLSLD_LO;
       break;
     case PPC::ADDItocL8:
+      // Skip the following peephole optimizations for ADDItocL8 on AIX which
+      // is used for toc-data access.
+      if (Subtarget->isAIXABI())
```

Comment:
There is a follow on patch for adding this peephole optimization: https://github.com/llvm/llvm-project/pull/76488 since earlier patches related to this were reverted as the code needed some refactoring. 
See: https://reviews.llvm.org/D101470 and  https://reviews.llvm.org/D155957